### PR TITLE
📖Added KubeVirt to the providers list

### LINKED ISSUE #5918: 📖Add KubeVirt to providers list
Now that we have Cluster Api Provider KubeVirt, that should be added to the providers list [here](https://cluster-api.sigs.k8s.io/reference/providers.html)
To make the changes, add
Kubevirt with link redirecting to the repository: https://github.com/kubernetes-sigs/cluster-api-provider-kubevirt

/kind documentation

### PR DESCRIPTION
**Pull request by MukulKolpe**
**What this PR does / why we need it**:
This PR adds KubeVirt to the providers list.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #5918

### CODE REVIEW

File: docs/book/src/reference/providers.md [hunk 1]
```diff
@@ -29,6 +29,7 @@ updated info about which API version they are supporting.
 - [Sidero](https://github.com/talos-systems/sidero)
 - [Tencent Cloud](https://github.com/TencentCloud/cluster-api-provider-tencent)
 - [vSphere](https://github.com/kubernetes-sigs/cluster-api-provider-vsphere)
+- [KubeVirt](https://github.com/kubernetes-sigs/cluster-api-provider-kubevirt)
```
Comment (fabriziopandini) [inline docs/book/src/reference/providers.md:32]: Providers are in alphabetical order. could you move Kubevirt up in the list?

Comment (MukulKolpe) [reply docs/book/src/reference/providers.md]: Thank you @fabriziopandini. I've made the required changes. Can you help to solve the recent failed test?

**Comment by sbueringer** [comment]
@MukulKolpe Thx, can you please squash the commits?

**Review by fabriziopandini — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**